R/makeTreeSummarizedExperimentFromqiime2.R

```diff
@@ -215,10 +235,20 @@ parse_q2taxonomy <- function(taxa, sep = "; |;", trim_rank_prefix = TRUE) {
         taxa$Taxon <- gsub("D_\\d__", "", taxa$Taxon)
     }
 
+    #  work with any combination of taxonomic ranks available
+    all_ranks <- c("Kingdom","Phylum","Class","Order","Family","Genus","Species")
+    all_prefixes <- c("k__", "p__", "c__", "o__", "f__", "g__", "s__")
+    present_ranks <- strsplit(taxa[, "Taxon"], sep)
```

Comment:
This information is not present anymore, if `trim_rank_prefix = TRUE`.

So this does not work as intended I guess?